src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2083,14 +2090,30 @@ protected override void VisitCatchBlock(BoundCatchBlock catchBlock, ref LocalSta
         {
             if (trackUnassignments)
             {
+#if REFERENCE_STATE
+                LocalState oldTryState = _tryState;
```

Comment:
instead of adding all these ```#if```, it would seem simpler to have a local FakeNullable class with a Value field, just to hold the try state.